postreise/plot/plot_states.py

```diff
@@ -64,157 +66,245 @@ def get_state_borders():
+    """Add legend.
+
+    :param bokeh.plotting.figure canvas: canvas.
+    :param dict state2label: keys are states abbreviation and values are labels.
+    :param str title: title for legend.
+    :param str location: legend location on canvas. Default is bottom right.
+    :param str title_size: legend title font size. Default is 12pt.
+    :param str label_size: legend labels font size. Default is 12pt.
+    :return: (*bokeh.plotting.figure.Figure*) -- canvas with colored state.
+    :raises TypeError:
+        if ``title`` is not None or str.
+        if ``location`` is not a str.
+        if ``title_size`` is not a str.
+        if ``label_size`` is not a str.
+    """
+    if title is not None and not isinstance(title, str):
+        raise TypeError("title must be a str")
+    if location is not None and not isinstance(location, str):
+        raise TypeError("location must be a str")
+    if title_size is not None and not isinstance(title_size, str):
+        raise TypeError("title_size must be a str")
+    if label_size is not None and not isinstance(label_size, str):
+        raise TypeError("label_size must be a str")
+
+    patches = canvas.select_one({"name": "states"})
+    patches = expand_data_source(patches, state2label, "state_legend")
+
+    group_legend = (
+        pd.DataFrame({"legend": patches.data_source.data["state_legend"]})
+        .groupby(["legend"])
+        .groups
+    )
+
+    legend = Legend(
+        items=[
+            LegendItem(index=i[0], label=l, renderers=[patches])
+            for l, i in group_legend.items()
+        ]
+    )
+    canvas.add_layout(legend)
+
+    if title is not None:
+        canvas.legend.title = title
+    canvas.legend.location = location
+    canvas.legend.label_text_font_size = label_size
+    canvas.legend.title_text_font_size = title_size
+
+    return canvas
+
+
+def plot_states(
+    figsize=(1400, 800),
+    state_borders_kwargs=None,
+    state_colors_args=None,
+    state_tooltips_args=None,
+    state_legends_kwargs=None,
+):
+    """Add states on canvas and optionally add colors, tooltips and legends.
+
+    :param tuple figsize: size of the bokeh figure (in pixels).
+    :param dict state_borders_kwargs: keyword argument(s) to be passed to
+        :func:`postreise.add_state_borders`.
+    :param dict state_colors_args: arguments to be passed to
+        :func:`postreise.add_state_colors`.
+    :param dict state_tooltips_args: argument(s) to be passed to
+        :func:`postreise.add_state_tooltips`.
+    :param dict state_legends_kwargs: keyword argument(s) to be passed to
+        :func:`postreise.add_state_legends`.
+    :raises TypeError:
+        if ``state_borders_kwargs`` is not None or dict.
+        if ``state_colors_args`` is not None or dict.
+        if ``state_tooltips_args`` is not None or dict.
+        if ``state_legends_kwargs`` is not None or dict.
+    :return: (*bokeh.plotting.figure.Figure*) -- map of us states.
+    """
+    for var, label in [
```

Comment:
The conversation about `eval` could apply here, should we end up using that

Reply:
Done